pkg/modeling/modeling.go

```diff
@@ -14,15 +13,11 @@ import (
 	clusterapis "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
 )
 
-var (
-	mu                  sync.Mutex
-	defaultModelSorting []clusterapis.ResourceName
```

Comment:
The resource models might be different from each other, for example, cluster 1's model could be:
```yaml
resourceModels:
  - grade: 0
    ranges:
    - max: "1"
      min: "0"
      name: cpu        # Note that only declared cpu
```

and cluster 2's model could be
```yaml
resourceModels:
  - grade: 0
    ranges:
    - max: 4Gi
      min: "0"
      name: memory   # Note that only declared memory
```

The global variable `defaultModelSorting` actually shared by different clusters and might be overwritten during init the model at https://github.com/karmada-io/karmada/blob/3267155766f69083d211c4bc804e8bda4b18fd4d/pkg/controllers/status/cluster_status_controller.go#L595C35-L595C46.

@halfrost Please correct me if this is not the case.

Reply:
Yes, the resource model should be different for each cluster. But the original requirement seems to be that this modeling is to build a large resource model. Karmada manages the resource models of all clusters in a unified way. For this reason, it is taken out as a global variable.